src/odb.c

```diff
@@ -1012,8 +1018,10 @@ static int odb_read_1(git_odb_object **out, git_odb *db, const git_oid *id,
 	bool found = false;
 	int error = 0;
 
-	if (!only_refreshed && odb_read_hardcoded(&raw, id) == 0)
-		found = true;
+	if (!only_refreshed) {
+		if ((error = odb_read_hardcoded(&found, &raw, id)) < 0)
+			return error;
```

Comment:
Shouldn't this rather be a `goto out`? While we obviously have nothing to clean up, it'd be a readability improvement and more consistent to our usual patterns.

Reply:
I don't think so; this was intentional.  Doing that relies on knowing that `odb_read_hardcoded` will always exit having set `raw.data` to either a pointer that can be freed, or `NULL`.  This feels brittle, and if we change `odb_read_hardcoded` in the future to have another check and `return -1` before the allocation of `raw.data`, then we would try to dereference that in the `free`.

Reply:
Don't we assume this in most of our code when calling functions which have out-parameters? They either don't touch the out parameter at all in case of an error or sanitize it to always contain a value which can be handled.

Reply:
Yes - and here we're assuming that it doesn't touch it at all.

Reply:
We could have a `goto out` here if we changed the declaration to set the members to `NULL`.  I don't think that's a big win for readability, but if you do, then I can make that change.

Reply:
I don't want to be too nitpicky. Just go with your current style ;)